src/main/kotlin/org/rust/ide/inspections/RsUnresolvedReferenceInspection.kt

```diff
@@ -96,12 +112,8 @@ class RsUnresolvedReferenceInspection : RsLocalInspectionTool() {
         val fixes = createQuickFixes(candidates, element, context)
 
         val highlightedElement = element.referenceNameElement ?: element
-        registerProblem(
-            highlightedElement,
-            description,
-            ProblemHighlightType.LIKE_UNKNOWN_SYMBOL,
-            *fixes.toTypedArray()
-        )
+        val highlightType = if (context?.type == AutoImportFix.Type.PAT_BINDING) INFORMATION else LIKE_UNKNOWN_SYMBOL
+        registerProblem(highlightedElement, description, highlightType, *fixes.toTypedArray())
```

Comment:
This code contains following problems:
- you always register a problem with `INFORMATION` hightlighting type, even in batch mode. In leads to
```
com.intellij.diagnostic.PluginException: Tool #RsUnresolvedReference (class org.rust.ide.inspections.RsUnresolvedReferenceInspection) registers 'INFORMATION'-level problem in batch mode on FILE. Warnings of the 'INFORMATION' level are invisible in the editor and should not become visible in batch mode. Moreover, since 'INFORMATION'-level fixes act more like intention actions, they could e.g. change semantics and thus should not be suggested for batch transformations [Plugin: org.rust.lang]
	at com.intellij.codeInspection.InspectionEngine$2.registerProblem(InspectionEngine.java:261)
	at com.intellij.codeInspection.ProblemsHolder.registerProblem(ProblemsHolder.java:52)
	at org.rust.ide.inspections.RsProblemsHolder.registerProblem(RsLocalInspectionTool.kt:86)
	at org.rust.ide.inspections.RsUnresolvedReferenceInspection.registerProblem(RsUnresolvedReferenceInspection.kt:116)
	at org.rust.ide.inspections.RsUnresolvedReferenceInspection.access$registerProblem(RsUnresolvedReferenceInspection.kt:23)
	at org.rust.ide.inspections.RsUnresolvedReferenceInspection$buildVisitor$1.visitPatBinding(RsUnresolvedReferenceInspection.kt:78)
	at org.rust.lang.core.psi.impl.RsPatBindingImpl.accept(RsPatBindingImpl.java:21)
	at org.rust.lang.core.psi.impl.RsPatBindingImpl.accept(RsPatBindingImpl.java:26)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:56)
	at com.intellij.codeInspection.InspectionEngine.lambda$inspectElements$9(InspectionEngine.java:282)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$2(JobLauncherImpl.java:140)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$3(JobLauncherImpl.java:136)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:915)
	at com.intellij.concurrency.JobLauncherImpl.processImmediatelyIfTooFew(JobLauncherImpl.java:147)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:47)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:51)
	at com.intellij.codeInspection.InspectionEngine.lambda$inspectElements$10(InspectionEngine.java:301)
	at com.intellij.codeInspection.InspectionEngine.withSession(InspectionEngine.java:226)
	at com.intellij.codeInspection.InspectionEngine.inspectElements(InspectionEngine.java:242)
	at com.intellij.codeInspection.InspectionEngine.inspectElements(InspectionEngine.java:215)
	at com.intellij.codeInspection.InspectionEngine.inspectEx(InspectionEngine.java:121)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$inspectFile$19(GlobalInspectionContextImpl.java:493)
	at com.intellij.codeInspection.ex.InspectionProfileWrapper.runWithCustomInspectionWrapper(InspectionProfileWrapper.java:89)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.inspectFile(GlobalInspectionContextImpl.java:490)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$runTools$6(GlobalInspectionContextImpl.java:353)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$0(DumbService.java:113)
	at com.intellij.openapi.project.DumbService.lambda$runReadActionInSmartMode$1(DumbService.java:157)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:941)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:150)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:113)
	at com.intellij.openapi.project.DumbService.tryRunReadActionInSmartMode(DumbService.java:130)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.lambda$runTools$12(GlobalInspectionContextImpl.java:338)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.lambda$call$0(JobLauncherImpl.java:297)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.call(JobLauncherImpl.java:282)
	at com.intellij.concurrency.JobLauncherImpl.processQueue(JobLauncherImpl.java:350)
	at com.intellij.codeInspection.ex.GlobalInspectionContextImpl.runTools(GlobalInspectionContextImpl.java:394)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase.lambda$performInspectionsWithProgress$2(GlobalInspectionContextBase.java:273)
	at com.intellij.psi.impl.PsiManagerImpl.runInBatchFilesMode(PsiManagerImpl.java:451)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase.lambda$performInspectionsWithProgress$3(GlobalInspectionContextBase.java:267)
	at com.intellij.codeInspection.reference.RefManagerImpl.runInsideInspectionReadAction(RefManagerImpl.java:495)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase.performInspectionsWithProgress(GlobalInspectionContextBase.java:266)
	at com.intellij.codeInspection.ex.GlobalInspectionContextBase$2.run(GlobalInspectionContextBase.java:227)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$5(CoreProgressManager.java:493)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
```
- we provide `Unresolved reference` description although reference is not unresolved. Currently, it doesn't affect users because the corresponding message is not shown but anyway
- if `fixes` is empty, there is no reason to register anything at all